http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/websocket/message/WebSocketBinaryMessageImpl.java

```diff
@@ -0,0 +1,66 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.carbon.transport.http.netty.contractimpl.websocket.message;
+
+import org.wso2.carbon.transport.http.netty.contract.websocket.WebSocketBinaryMessage;
+import org.wso2.carbon.transport.http.netty.contractimpl.websocket.WebSocketMessageContextImpl;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Implementation of {@link WebSocketBinaryMessage}.
+ */
+public class WebSocketBinaryMessageImpl extends WebSocketMessageContextImpl implements WebSocketBinaryMessage {
+
+    private final ByteBuffer buffer;
+    private final boolean isFinalFragment;
+
+    public WebSocketBinaryMessageImpl(ByteBuffer buffer, boolean isFinalFragment,
+                                      WebSocketMessageContextImpl channelContext) {
+        super(channelContext.getChannelSession(), channelContext.getChannelContext());
+        this.buffer = buffer;
+        this.isFinalFragment = isFinalFragment;
+    }
+
+    @Override
+    public ByteBuffer getByteBuffer() {
+        return buffer;
+    }
+
+    @Override
+    public byte[] getByteArray() {
+        byte[] bytes;
+        if (buffer.hasArray()) {
+            return buffer.array();
```

Comment:
To be consistent with the flow, make this "bytes = buffer.array()", since a "bytes" variable is declared in the first line, and its returned in the end.